advocacy_docs/edb-postgres-ai/migration-etl/data-migration-service/getting_started/prepare_schema.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: "Preparing and importing the schema"
+deepToC: true
+---
+
+Before you use EDB Data Migration Service (EDB DMS) to configure a data migration, you must prepare and import your schema to the target database. 
+
+Some of your schema's constraints must be included before the data migration takes place, whereas others must be applied [after the data migration is completed](apply_constraints). This ensures you can migrate without performance degradation. 
+
+## Schema integrity and performance considerations
+
+The presence of target database constraints, triggers, and WAL logging can impact the data migration performance. When possible, EDB recommends a two-step import of schema constraints.
+
+### `PRIMARY KEY` and `UNIQUE` constraints
+
+`PRIMARY KEY` and `UNIQUE` constraints are leveraged by EDB DMS to provide an exactly-once delivery when migrating data to the target database. Therefore, `PRIMARY KEY` and `UNIQUE` constraints should be included in the schema import that you perform before the data migration begins. Other types of constraints should be excluded from the schema import. 
+
+For rows in tables that do not have `PRIMARY KEY` or `UNIQUE` constraints it is only possible to achieve at-least-once delivery. Deduplication can be performed during the [data migration verification](verify_migration).
+
+!!!note 
+   `NOT NULL` constraints don't represent a significant performance impact for destination servers and can also be included in the schema import.
+!!!
+
+### `FOREIGN KEY`, `REFERENCES`, `CHECK`, `CASCADE` and `EXCLUDE` constraints
+
+EDB DMS is able to apply change events in parallel against destination database clusters. However, migrating some constraint types can negatively affect the performance of the migration. These type of constraints lead to unnecessary CPU and memory utilization in the context of an in-flight data migration from a consistent and referentially integral source database.
+
+EDB recommends [applying the following constraints](apply_constraints) on the target database after you have signalized the end of the CDC stream by [marking the migration as completed](mark_completed) in the Console.
+
+`FOREIGN KEY` / `REFERENCES`
+
+`CHECK`
+
+`CASCADE`
+
+`EXCLUDE` 
```

Comment:
@mlewandowski-edb - Wanted to check with you on this: I checked out the postgres docs you cited above, but I saw `EXCLUDE USING` being used there but I didn't see `EXCLUSION` anywhere. I did see "exclusion constraint" being used there. Will it confuse readers to use `EXCLUSION` as such if not used in Postgres's docs? I mean you can't use the KEYWORD `EXCLUSION` in Postgres can you?

Reply:
It seems calling them EXCLUDE constraints rather than EXCLUSION constraints would be more consistent with the PostgreSQL syntax and less likely to cause confusion. I think it's important to keep terminology aligned with the actual SQL language and syntax when documenting or discussing these constraints.

Reply:
I don't agree. `EXCLUDE` is a keyword used to create an `EXCLUSION` constraint not the name of the type of a constraint. The PostgreSQL documentation does refer to them as "`EXCLUSION` Constraints" as evidenced by the title of this section:
https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-EXCLUSION

Reply:
But the section refers to the as "Exclusion Constraints" not "EXCLUSION Constraints", putting the word exclusion in all caps is what I am having an issue with. Presenting it in all caps is presenting as a keyword. But it is not a keyword, so it could easily confuse the reader.

What if we just take the caps out of it and call it "exclusion constraints?", as this is in line with the documentation and not confusing as EXCLUSION constraints, which being in all caps suggests it is a keyword?

Reply:
I see. Yes. Let's just call them exclusion (or Exclusion, when appropriate) constraints.  For that matter, you could simply call the other constraints foreign key and check constraints without stylizing them as keywords.

Reply:
Got it. Agreed. Thanks Matt.